src/visitor.rs

```diff
@@ -401,3 +400,72 @@ impl<'input, V: InnerListVisitor<'input>> InnerListVisitor<'input> for Option<V>
         }
     }
 }
+
+/// A visitor that cannot be instantiated, but can be useds a type in situations
+/// guaranteed to return an error `Result`, analogous to
+/// [`std::convert::Infallible`].
+#[derive(Clone, Copy)]
+pub enum Never {}
+
+impl<'input> ParameterVisitor<'input> for Never {
+    type Error = Infallible;
+
+    fn parameter(
+        &mut self,
+        _key: &'input KeyRef,
+        _value: BareItemFromInput<'input>,
+    ) -> Result<(), Self::Error> {
+        match *self {}
```

Comment:
It seems to me like this should be instead:

```suggestion
        unreachable!()
```

...throughout.

Either that, or this should implement `Error` and these functions should all return `Err(Self)`.

Reply:
Unfortunately, `unreachable!()` doesn't suffice for the ones using `impl Trait`. CC https://github.com/rust-lang/rust/issues/36375.

Reply:
Yeah, I solved that in the most unpleasant manner possible in my code:

```suggestion
        unreachable!("message");
        #[allow(whatever_comes)]
        Ok(Ignored) // or whatever
```